lib/srv/authhandlers.go

```diff
@@ -521,14 +553,35 @@ func (h *AuthHandlers) IsHostAuthority(cert ssh.PublicKey, address string) bool
 	return true
 }
 
+// loginChecker checks if the Teleport user should be able to login to
+// a node. It is created for tests.
+type loginChecker interface {
+	// canLoginWithoutRBAC checks the given certificate (supplied by a connected
+	// client) to see if this certificate can be allowed to login as user:login
+	// pair to requested server.
+	canLoginWithoutRBAC(cert *ssh.Certificate, clusterName string, teleportUser string, osUser string) error
+	// canLoginWithRBAC checks the given certificate (supplied by a connected
+	// client) to see if this certificate can be allowed to login as user:login
+	// pair to requested server and if RBAC rules allow login.
+	canLoginWithRBAC(cert *ssh.Certificate, clusterName string, node types.Server, teleportUser, osUser string) error
+}
+
+type ahLoginChecker struct {
+	h *AuthHandlers
```

Comment:
Do we need `AuthHndlers` struct to just use a logger and one function? Can we reduce the requirements here?

Reply:
Updated so only logger, config and small interface is needed, does it look good now?